src/handler/reactionHandler.js

```diff
@@ -21,41 +22,79 @@ const voteEmojis = ["1️⃣", "2️⃣", "3️⃣", "4️⃣", "5️⃣", "6️
         const member = message.guild.members.cache.get(client.users.cache.get(data.user_id).id);
-        if (
-            member.id === client.user.id ||
-            message.embeds.length !== 1 ||
-            message.embeds[0].author.name.indexOf("Strawpoll") < 0 && voteEmojis.slice(0, 10).includes(event.d.emoji.name)
-        ) return;
-
-        const reactions = message.reactions.cache.filter(reaction =>
-            reaction.users.cache.has(member.id) &&
-            reaction._emoji.name !== event.d.emoji.name &&
-            voteEmojis.includes(reaction._emoji.name)
-        );
-
-        for (let reaction of reactions.values()) await reaction.users.remove(member.id).catch(log.error);
+        const fromThisBot = member.id === client.user.id;
+        const isStrawpoll = message.embeds.length === 1 &&
+            message.embeds[0].author.name.indexOf("Strawpoll") >= 0 && voteEmojis.slice(0, 10).includes(event.d.emoji.name);
+
+        const isUmfrage = message.embeds.length === 1 &&
+            message.embeds[0].author.name.indexOf("Umfrage") >= 0 && voteEmojis.slice(0, 10).includes(event.d.emoji.name);
+
+
+        if(fromThisBot) {
+            return;
+        }
+
+        const delayedPoll = poll.delayedPolls.find(x => x.pollId === message.id);
+        const isDelayedPoll = Boolean(delayedPoll);
+
+        if(isStrawpoll) {
+            if(isDelayedPoll) {
+                delayedPoll.reactions.forEach(reactionList => {
+                    reactionList.forEach((x, i) => {
+                        if(x === member.id) reactionList.splice(i);
+                    });
+                });
+                const delayedPollReactions = delayedPoll.reactions[voteEmojis.indexOf(event.d.emoji.name)];
+                delayedPollReactions.push(member.id);
+            }
+
+            const reactions = message.reactions.cache.filter(reaction =>
+                reaction.users.cache.has(member.id) &&
+                reaction._emoji.name !== event.d.emoji.name &&
+                voteEmojis.includes(reaction._emoji.name)
+            );
+
+            for (let reaction of reactions.values()) await reaction.users.remove(member.id).catch(log.error);
+        }
+        else if(isUmfrage) {
+            if(isDelayedPoll) {
+                const delayedPollReactions = delayedPoll.reactions[voteEmojis.indexOf(event.d.emoji.name)];
+                if(!delayedPollReactions.some(x => x === member.id)) {
+                    delayedPollReactions.push(member.id);
+                }
+            }
+        }
+
+        // If it's a delayed poll, we clear all Reactions
+        if(isDelayedPoll) {
+            const allUserReactions = message.reactions.cache.filter(reaction =>
+                reaction.users.cache.has(member.id) &&
+                voteEmojis.includes(reaction._emoji.name)
+            );
+            for (let reaction of allUserReactions.values()) await reaction.users.remove(member.id).catch(log.error);
```

Comment:
Isn't it better to just not add the reactions in the first place?

Reply:
Only the reactions of the user that reacted will get deleted -> We can't supress the adding that discord itself does

Reply:
okay tru tru